src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/FrameResponseStream.cs

```diff
@@ -31,6 +34,7 @@ public override long Position
             set => throw new NotSupportedException();
         }
 
+        // REVIEW: Should Flush() throw?
```

Comment:
Kestrel's flush needs to throw for all the same reasons write is throwing, sync over async. HttpSys's flush no-ops unless you haven't sent the headers, so throwing could be conditional there but that would be even more confusing.